### DISCUSSION TITLE
FPP Remote on DigQuad

### DISCUSSION THREAD

**Discussion by GeoTime61** [forkineye/ESPixelStick#706] [Q&A]
I am trying to run a DigQuad ESP32 controller with an SD card as an FPP Remote. From xLights I can upload the controller configuration and use FPP Connect to upload sequence files. The files show up on the DigQuad and FPP sees the DigQuad as a Remote device. However, when FPP plays a sequence the LEDs do not turn on. If I take the DigQuad out of remote mode and send DDP data to it directly from FPP everything works so I think it is hooked up correctly. I am using FPP version 7.4 and ESPixelStick version 4.0-beta5. Has anyone had success with a DigQuad or other ESP32 controller using FPP Remote?

**Comment by MartinMueller2003** [comment]
Many people have had success and many people have had issues. Rather than play the sequences via DDP. play them via the UI. On the UI, change the FPP Remote selection to the first file in your show. If nothing comes out, then xLights had an issue uploading the fseq file (this happens a lot). If the file plays, then put the ESP back in remote mode and start playing on FPP. Then look at the ESP status page. Is it showing the IP Address of the FPP? Do the statistics show that the ESP is receiving sync messages?

You can also go on the admin page and select "Advanced" mode. Then return to the status page. Under the FPP statistics, there will now be a "Last error" message. Is it blank? If not, what is the message?

**Reply by GeoTime61** [reply]
I changed FSEQ File To Play to one of the sequences on the SD card. When I clicked Save Changes the ESPS rebooted. The changes survived the crash/reboot and the Home page shows the Time Elapsed counting up, but there is no output on the lights. I tried FPP Connect several times, never saw any errors, but never got any LED output. I went back to FPP Remote mode and see the Sync Packets counting up, the Master IP address is correct, but no output to the LEDs. Last Error is blank. After about ten seconds the ESPS software freezes and I lose the network connection. Any other settings I can try?

**Reply by GeoTime61** [reply]
This crashes 100% of the time for me: Copy an uncompressed .fseq file to the ESPS board. Set Primary Input to Disabled and Secondary Input to FPP Remote. Set one Output to WS2811. Save Changes. Change FSEQ File To Play to the .fseq file that was uploaded. Save Changes. System will crash and reboot. Any Save Changes will crash again until you change FSEQ File To Play to Play Remote Sequence.

**Comment by MartinMueller2003** [comment]
Leave the ESPS connected to the flash tool. clear the logs, play a file using the UI, save the log from the flashtool and send it to me as an attachment.

**Reply by GeoTime61** [reply]
Log file after the ESPS plays the .fseq file for a few seconds.

[14-dec-23 1704.txt](https://github.com/forkineye/ESPixelStick/files/13686927/14-dec-23.1704.txt)

**Comment by MartinMueller2003** [comment]
Can you give me the fseq file that caused the reboot while it was running.

**Reply by GeoTime61** [reply]
[Remote Test.fseq.zip](https://github.com/forkineye/ESPixelStick/files/13679770/Remote.Test.fseq.zip)

**Comment by MartinMueller2003** [comment]
That is a V2.2 compressed file. It cannot be played on the ESP V4 SW. ESP only plays uncompressed "Sparse" files.

**Reply by MartinMueller2003** [reply]
I am seeing two bugs here. 

1. Saving the config file causes the ESP to reboot.
2. Unsupported compressed file error message is not reported to the UI.

**Reply by GeoTime61** [reply]
Interesting, because xLights FPP Connect shows the format as V2 Sparse/Uncompressed.

**Reply by GeoTime61** [reply]
<img width="1222" alt="Screenshot 2023-12-11 at 2 28 49 PM" src="https://github.com/forkineye/ESPixelStick/assets/102134695/0bc1d91a-55c4-40cc-82a7-4a14689adfeb">

**Reply by GeoTime61** [reply]
Another issue is that sometimes after configuring the Secondary Input and Output fields, the user interface does not show the additional FPP Remote or Output fields. After a reboot they show up correctly. 
![Screenshot 2023-12-14 at 5 06 42 PM](https://github.com/forkineye/ESPixelStick/assets/102134695/3590442c-8a30-4735-a0d9-4ee9953135f1)

**Reply by GeoTime61** [reply]
I changed the xLights global Preferences to V2 Uncompressed, re-rendered the file, and uploaded it via FPP Connect. The file size did not change on the ESPS. I deleted the file and uploaded again, but still had the smaller file size. I deleted again and used and drag and drop to upload the file. This time I got the larger, uncompressed file size. Not sure what is going on there.

**Reply by GeoTime61** [reply]
I left the ESPS board running all day, then changed FPP Remote Configuration to play the (now definitely uncompressed) .fseq file and as soon as I hit Save Changes, the board rebooted and lost my Output configuration which had been saved previously. It remembered the Secondary Input and FPP Remote Configuration settings.

**Reply by GeoTime61** [reply]
The timestamp on all uploaded files shows as 12/31/1979. Is there a way to get the date/time updated? The ESPS board is connected via Wi-Fi to a router which is connected to the Internet. xLights shows the correct date and time on the .fseq files.

**Comment by MartinMueller2003** [comment]
Looks like you have found an xlights bug as well.

**Comment by MartinMueller2003** [comment]
the ESP gets its time from a browser. NTP is not currently supported. Many show networks do not have an internet connection and cannot use NTP. The show player relies on an RTC.

**Reply by GeoTime61** [reply]
I'm not sure what you mean by "gets its time from a browser." I am connected to the ESPS board through Google Chrome on my computer which has Internet access and the correct time. If I drag and drop a file onto the File Management screen, the date is always 12/31/1979.

**Comment by MartinMueller2003** [comment]
I am actually seeing three crash modes. I have fixed one and am working on the others.

**Reply by GeoTime61** [reply]
Excellent - thank you! I'm happy to test any new versions that you want feedback on.

**Comment by MartinMueller2003** [comment]
New dist.zip has been uploaded to the google drive and my github repository.. It should fix most of your issues.
    On Saturday, December 16, 2023 at 09:19:31 PM EST, GeoTime61 ***@***.***> wrote:  
 
 


Excellent - thank you! I'm happy to test any new versions that you want feedback on.

**Reply by GeoTime61** [reply]
I downloaded the dist.zip with Build Date Dec 18 2023 - 18:58:44. 

I set Primary Input to Disabled and Secondary Input to FPP Remote. Set one Output to WS2811. Save Changes. The Up Time counter on the Home page resets to 00:00. 

Change FSEQ File To Play to one of the downloaded V2 Sparse/Uncompressed files. Save Changes. The sequence plays correctly.

Change FSEQ File To Play to Play Remote Sequence. Save Changes. FPP Remote Status shows the FPP Master IP address. Use xLights FPP Connect to download a new sequence file. Play the sequence from FPP. The ESPS Home page shows both DDP Status and FPP Remote Status and the DDP Packets are counting up. I wasn't expecting this because I had disabled Primary Input. The Device Setup page shows Primary Input as DDP. This must have changed because of the FPP Connect download. Set Primary Input to Disabled. Save Changes. The Home page Up Time counter resets to 00:00 and the DDP Status goes away. Play the sequence from FPP and it plays correctly.

Huge improvement - thank you!

**Comment by MartinMueller2003** [comment]
A reboot is required whenever the input type is changed. Most of the libraries used do not handle turning off the protocol. Same for turning off an RMT output. I had to use libraries that require a reboot to turn them off.
